src/main/java/com/microsoft/jfr/RecordingOptions.java

```diff
@@ -0,0 +1,385 @@
+     * Get the maximum length of time that data is kept on disk before it <em>may</em>
+     * be deleted. The return value will be {@code "0""} if the option has not been set.
+     * @return The {@code maxAge} recording option.
+     */
+    public String getMaxAge() {
+        return recordingOptions.getOrDefault(Option.MAX_AGE.name, Option.MAX_AGE.defaultValue);
+    }
+
+    /**
+     * Get the maximum size, in bytes, at which data is kept on disk. The return
+     * value will be {@code "0"} if the option has not been set.
+     * @return The {@code maxSize} recording option.
+     */
+    public String getMaxSize() {
+        return recordingOptions.getOrDefault(Option.MAX_SIZE.name, Option.MAX_SIZE.defaultValue);
+    }
+
+    /**
+     * Get whether to dump recording data to disk when the JVM exits.
+     * @return {@code "true"} if recording data is dumped to disk on JVM exit.
+     */
+    public String getDumpOnExit() {
+        return recordingOptions.getOrDefault(Option.DUMP_ON_EXIT.name, Option.DUMP_ON_EXIT.defaultValue);
+    }
+
+    /**
+     * Get the path to where recording data is written. The path is relevant to the
+     * machine on which the JVM is running. The return value will be an empty String
+     * if the option has not been set.
+     * @return The path to where recording data is written.
+     */
+    public String getDestination() {
+        return recordingOptions.getOrDefault(Option.DESTINATION.name, Option.DESTINATION.defaultValue);
+    }
+
+    /**
+     * Get whether to save flight recordings to disk.
+     * @return {@code "true"} if flight recordings are saved to disk.
+     */
+    public String getDisk() {
+        return recordingOptions.getOrDefault(Option.DISK.name, Option.DISK.defaultValue);
+    }
+
+    /**
+     * Get how long the recording should be running. The return value will be
+     * {@code "0"} if the option has not been set.
+     * @return The {@code duration} for a recording.
+     */
+    public String getDuration() {
+        return recordingOptions.getOrDefault(Option.DURATION.name, Option.DURATION.defaultValue);
+    }
+
+    /**
+     * Get the read-only recording options. The keys are names of recording options
+     * according to FlightRecorderMXBean.
+     * @return A read-only map of the recording options.
+     */
+    /*package*/ Map<String,String> getRecordingOptions() {
+        return recordingOptions;
+    }
+
+    // The configuration options. The keys are names of recording options
+    // according to FlightRecorderMXBean.  The value is a valid value for
+    // the option. Options that take on default values should be absent.
+    private final Map<String,String> recordingOptions;
+
+    // format for FlightRecorderMXBean maxAge and duration recording options
+    private static final Pattern pattern = Pattern.compile("([-+]?\\d+)\\s*(\\w*)");
+
+    // If the durationString arg is a valid format, return the arg properly formatted for FlightRecorderMXBean.
```

Comment:
Add a comment as to why you didn't use the Duration class (which is totally fair).